### PR TITLE
First pass at improving local development tutorial

### LINKED ISSUE #1434: Update "Developing Slack apps locally" document page to mention Socket Mode
"Developing Slack apps locally" document page should mention Socket Mode.

### List any relevant URLs

- https://slack.dev/node-slack-sdk/tutorials/local-development
- https://github.com/slackapi/node-slack-sdk/blob/main/docs/_tutorials/local_development.md

### Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

### PR DESCRIPTION
**Pull request by filmaj**
###  Summary

This fixes #1434.

As discussed in #1434, linking the app and token creation parts to the bolt-js tutorial to try to consolidate into one location. Also split the guide up into two sections: one dedicated for HTTP (which mostly re-uses the existing guide), and added a new section calling out Socket Mode (and linking to the existing socket mode instructions). Finally, removed references to deprecated packages.

### CODE REVIEW

File: docs/_tutorials/local_development.md [hunk 1]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
+2. [HTTP](https://api.slack.com/apis/connections/events-api). Expose your app using a public-facing URL that Slack will send HTTP POST requests to. A bit harder to set up, but more resilient to network disruptions.
 
-![OAuth Tokens](bot-token.png "OAuth Tokens")
+This tutorial will cover how to set up your app using either approach.
 
-You'll see two tokens. To add scopes to the `xoxp` tokens, you can scroll down to the **Scopes** section. For now, we'll just use the `xoxb` bot token.
+### Socket Mode
 
-You'll need your token to use any of the [Web API methods](https://api.slack.com/methods).
+Setting up your node.js application for Socket Mode development is fast and easy with the [`@slack/socket-mode`](https://slack.dev/node-slack-sdk/socket-mode) package. In the previous [Tokens and installing apps](#tokens-and-installing-apps) section, you should have created and saved an App Token. You'll need that when following the [socket mode package setup instructions](../socket-mode).
```
Comment (mwbrooks) [inline docs/_tutorials/local_development.md:42]: I could be wrong, but it looks like "Node.js" is [typically capitalized](https://en.wikipedia.org/wiki/Node.js):
```suggestion
Setting up your Node.js application for Socket Mode development is fast and easy with the [`@slack/socket-mode`](https://slack.dev/node-slack-sdk/socket-mode) package. In the previous [Tokens and installing apps](#tokens-and-installing-apps) section, you should have created and saved an App Token. You'll need that when following the [Socket Mode package setup instructions](../socket-mode).
```

File: docs/_tutorials/local_development.md [hunk 2]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
-## Setting up interactive components
-If you want to use features like buttons, select menus, datepickers, dialogs, or message actions, you'll need to enable interactivity.
+### HTTP
 
-> The `@slack/interactive-messages` package is used for Interactive Components
+#### What is a Request URL
+A Request URL is a public URL where Slack can send HTTP POST requests.
 
-Start by clicking **Interactive Components** on the left sidebar. Once you flip the switch, you'll see a textbox labeled **Request URL**.
+When a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send information about that event to your app's Request URL, like who clicked it in what channel.
```
Comment (mwbrooks) [inline docs/_tutorials/local_development.md:49]: 👌🏻 Great way to explain it!

File: docs/_tutorials/local_development.md [hunk 3]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
...
-
-The path and port values are optional. If your request URL includes a different path, you should specify it with `--path=/my/path/here` (no brackets). Similarly, if your development proxy is forwarding requests to a different port, you should specify it with `--port=8888` (no brackets). If you're using the defaults, you can ignore everything after `<signing_secret>`.
-
-Now that you've verified your endpoint, you can add events.
+Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-Request-URL), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.
```
Comment (mwbrooks) [inline docs/_tutorials/local_development.md:70]: If anyone else if curious, the `&amp;` in `Request URL Configuration &amp; Verification` renders correctly with Jekyll:
image

File: docs/_tutorials/local_development.md [hunk 4]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
-- [Setting up interactive components](#setting-up-interactive-components)
-  - [What is a Request URL?](#what-is-a-request-URL)
-  - [Using a local Request URL for development](#using-a-local-request-URL-for-development)
-  - [Add Request URL to your app configuration](#add-the-URL-to-your-app-configuration)
+- [Socket Mode vs. HTTP](#socket-mode-vs-http)
+  - [Socket Mode](#socket-mode)
+  - [HTTP](#http)
+    - [What is a Request URL?](#what-is-a-request-URL)
+    - [Using a local Request URL for development](#using-a-local-request-URL-for-development)
+    - [Add Request URL to your app configuration](#add-the-URL-to-your-app-configuration)
```
Comment (mwbrooks) [inline docs/_tutorials/local_development.md:20]: Testing locally, it looks like these anchors need to be lowercase to work:
```suggestion
    - [What is a Request URL?](#what-is-a-request-url)
    - [Using a local Request URL for development](#using-a-local-request-url-for-development)
    - [Add Request URL to your app configuration](#add-the-url-to-your-app-configuration)
```

File: docs/_tutorials/local_development.md (hunk 3 above)
Comment (mwbrooks) [inline docs/_tutorials/local_development.md:70]: Fixing anchors to be lowercase:
```suggestion
Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-request-url), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.
```

**Review by mwbrooks — APPROVED** [review]
Very nice improvement @filmaj! Thanks for taking this on. I tested it locally and found a few issues with the anchors needing to be lowercase. I've left suggestions where I noticed them.

**Comment by filmaj** [comment]
Thanks for the tweaks @mwbrooks !

File: docs/_tutorials/local_development.md [hunk 5]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
 
-Now that you have a bot user with permission to send messages to Slack, let's install the app to your workspace.
+## Socket Mode vs. HTTP
+Your app can [communicate with Slack using one of two methods](https://api.slack.com/apis/connections):
 
-Click **Install App** on the left sidebar and click the big **Install App to Workspace** button at the top of the page. Once you authorize the installation, you'll land on the **OAuth & Permissions** page.
+1. [Socket Mode](https://api.slack.com/apis/connections/socket). Connect to Slack using a direct and long-lived socket connection. While convenient, easy to setup and the recommended approach when getting started or iterating on your app, apps cannot use Socket Mode and be listed in the App Directory (for that you'll need to use [HTTP](#http)). It is also important to remember that Socket Mode is prone to network faults because the connection is long-lived.
+2. [HTTP](https://api.slack.com/apis/connections/events-api). Expose your app using a public-facing URL that Slack will send HTTP POST requests to. A bit harder to set up, but more resilient to network disruptions.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:36]:
```suggestion
2. [HTTP](https://api.slack.com/apis/connections/events-api). Expose your app using a public-facing URL that Slack will send HTTP requests to. This requires a few [additional steps](https://slack.dev/bolt-js/tutorial/getting-started-http#setting-up-events-with-http) to set up, but may be more resilient to network disruptions than communication via Socket Mode.  
```

File: docs/_tutorials/local_development.md [hunk 6]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
 
-You'll need your token to use any of the [Web API methods](https://api.slack.com/methods).
+Setting up your Node.js application for Socket Mode development is fast and easy with the [`@slack/socket-mode`](https://slack.dev/node-slack-sdk/socket-mode) package. In the previous [Tokens and installing apps](#tokens-and-installing-apps) section, you should have created and saved an App Token. You'll need that when following the [Socket Mode package setup instructions](../socket-mode).
 
-## Setting up interactive components
-If you want to use features like buttons, select menus, datepickers, dialogs, or message actions, you'll need to enable interactivity.
+### HTTP
 
-> The `@slack/interactive-messages` package is used for Interactive Components
+#### What is a Request URL
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:46]:
```suggestion
#### What is a Request URL?
```

File: docs/_tutorials/local_development.md [hunk 7]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
+A Request URL is a public URL where Slack can send HTTP POST requests.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:47]:
```suggestion
A Request URL is a public URL where Slack can send HTTP POST requests with information about events, interactions and other happenings inside your Slack workspace.
```

File: docs/_tutorials/local_development.md [hunk 8]
```diff
@@ -12,55 +12,43 @@ Feel free to skip around to the specific sections relevant to you — we won't m
...
 
-Start by clicking **Interactive Components** on the left sidebar. Once you flip the switch, you'll see a textbox labeled **Request URL**.
+When a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send information about that event to your app's Request URL, like who clicked it in what channel.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:49]:
```suggestion
For example, when a message is posted, a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send relevant information about that event to your app's Request URL such as the user who initiated the event, channel where the event occurred and other contextual details.
```

File: docs/_tutorials/local_development.md [hunk 9]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
 
 The output should show a generated URL that you can use (we recommend the one that starts with `https://`). This URL will be the base of your request URL, in this case `https://8e8ec2d7.ngrok.io`.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:60]:
```suggestion
The output should show a generated URL that you can use (Use the one that starts with `https://`). This URL will be the base of your request URL, in this case `https://8e8ec2d7.ngrok.io`.
```

File: docs/_tutorials/local_development.md [hunk 10]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
...
-Okay, so hopefully at this point you have some kind of public-facing URL. Now, let's add that to the **Interactive Components** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
-
-If you're using the default HTTP server in the `@slack/interactive-messages` package, you should append `/slack/actions` to your URL. For our example, this will be `https://8e8ec2d7.ngrok.io/slack/actions`. If you are using the Express middleware, you can set whichever path you like, just remember to make the path you mount the middleware into the application the same as the one you configure in Slack.
+#### Add the URL to your app configuration
+At this point you have a public-facing URL. Now, let's add that to the **Interactive Components** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
 
 ![Configuring a Request URL](request-url-config.png "Configuring a Request URL")
 
-## Listening to workspace events
+### Listening to workspace events
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:67]:
```suggestion
### Subscribing to workspace events
```
The instructions here seem more relevant to Subscribing your app to an event vs. setting up your event handler/listener code (for example in Bolt "listener" [means the handling code](https://slack.dev/bolt-js/concepts#event-listening)")

File: docs/_tutorials/local_development.md [hunk 11]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
...
 If you want your app to listen to events like when a reaction is added, when a user mentions your app, or [another Events API event](https://api.slack.com/events), you'll need to enable events for your app.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:68]:
```suggestion
If you want your app to be subscribed to workspace events (like when a reaction is added, when a user mentions your app, or [another Events API event](https://api.slack.com/events), you need to enable events for your app.
```

File: docs/_tutorials/local_development.md [hunk 12]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
 
 The output should show a generated URL that you can use (we recommend the one that starts with `https://`). This URL will be the base of your request URL, in this case `https://8e8ec2d7.ngrok.io`.
 
-### Add the URL to your app configuration
-Okay, so hopefully at this point you have some kind of public-facing URL. Now, let's add that to the **Interactive Components** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
-
-If you're using the default HTTP server in the `@slack/interactive-messages` package, you should append `/slack/actions` to your URL. For our example, this will be `https://8e8ec2d7.ngrok.io/slack/actions`. If you are using the Express middleware, you can set whichever path you like, just remember to make the path you mount the middleware into the application the same as the one you configure in Slack.
+#### Add the URL to your app configuration
+At this point you have a public-facing URL. Now, let's add that to the **Interactive Components** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:63]: Event Subscriptions and Interactivity and Shortcuts are both places where the Request URL needs to go. I've always felt that to be a really annoying experience for devs who may not understand the nuance/distinction between events generated from Interactions vs. events that fall under Event Subscriptions. More comments below to that effect. 
```suggestion
At this point you have a public-facing URL. Now, let's add that to the **Interactivity and Shortcuts** and **Events Subscriptions** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
```

File: docs/_tutorials/local_development.md [hunk 13]
```diff
@@ -71,40 +59,15 @@ Once you've installed a development proxy, run it to begin forwarding requests t
...
-
-Now that you've verified your endpoint, you can add events.
+Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-request-url), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.
 
 ### Adding event subscriptions
 On the **Event Subscriptions** page, you can add each event you want your app to subscribe to. In the tables below, you may add Workspace events and Bot events.
```
Comment (srajiang) [inline docs/_tutorials/local_development.md:73]: This section can be collapsed with the section above. Basically you toggle on Event Subscriptions and then you start adding events that you want to subscribe to.

**Review by srajiang — CHANGES_REQUESTED** [review]
Thanks for dusting these docs off @filmaj and giving them a polish. :) 

One change that we can consider with this PR is offer clarification on Slack Events vs. Interactions and Shortcuts and that Request URL needs to be provided to Slack in two different places depending on what feature they are trying to add. This might be an opportunity to just link to existing api.slack.com writeups on these features instead of trying to do that ourselves. Generally I tend to prefer to let api.slack.com docs pages explain the core Slack app model concepts, events vs interactions etc. 

I've left some comments with suggestions to that effect - edit modify as feels appropriate!

**Review by srajiang — APPROVED** [review]
Looking good!

**Pull request merged**